app/notifications/validators.py

```diff
@@ -99,17 +100,31 @@ def check_service_over_daily_message_limit(key_type: ApiKeyType, service: Servic
     counter_name="rate_limit.live_service_daily_sms",
     exception=LiveServiceTooManySMSRequestsError,
 )
-def check_service_over_daily_sms_limit_and_warn(key_type: ApiKeyType, service: Service):
-    if current_app.config["FF_SPIKE_SMS_DAILY_LIMIT"] and key_type != KEY_TYPE_TEST and current_app.config["REDIS_ENABLED"]:
-        fragments_sent = fetch_daily_sms_fragment_count(service.id)
-        warn_about_daily_sms_limit(service, fragments_sent)
+def check_sms_daily_limit(service: Service, key_type: ApiKeyType = KEY_TYPE_NORMAL):
+    if not current_app.config["FF_SPIKE_SMS_DAILY_LIMIT"]:
+        return
+    if key_type == KEY_TYPE_TEST:
+        return
+    if not current_app.config["REDIS_ENABLED"]:
+        return
+    _check_sms_daily_limit(service)
+
+
+def send_warning_sms_limit_emails_if_needed(service: Service, key_type: ApiKeyType = KEY_TYPE_NORMAL):
+    if not current_app.config["FF_SPIKE_SMS_DAILY_LIMIT"]:
+        return
+    if key_type == KEY_TYPE_TEST:
+        return
+    if not current_app.config["REDIS_ENABLED"]:
+        return
+    _send_warning_sms_limit_emails_if_needed(service)
 
 
 def check_rate_limiting(service: Service, api_key: ApiKey, template_type: TemplateType):
     check_service_over_api_rate_limit(service, api_key)
     check_service_over_daily_message_limit(api_key.key_type, service)
-    if template_type == "sms":
-        check_service_over_daily_sms_limit_and_warn(api_key.key_type, service)
+    if template_type == SMS_TYPE:
+        check_sms_daily_limit(api_key.key_type, service)
```

Comment:
This is causing the current test failures. The params are reversed, should be:
```suggestion
        check_sms_daily_limit(service, api_key.key_type)
```

Reply:
I gave the `key_type` argument a default value and made it optional, so that's why the order switched here. Arguably that is not worth the trouble.

Reply:
Right but the args are in the wrong order so this crashes.. unless I'm missing something.